### ISSUE TITLE
Dismissing transition (iOS 13) problems

### ISSUE THREAD

**Issue by uchiaTziki** [TimOliver/TOCropViewController#365]
I've encounter an issue when presenting the crop controller over a Navigation controller. 

When dismissing the crop controller, at the end of the transition the navigation controller view is being removed and not displayed... (happens in iPhone and iPad and only on iOS 13).

not to mention that the navigation is still there and the UI is "stuck".

**Comment by TimOliver** [comment]
@uchiaTziki Um. You might have to show me, sorry! Can you attach a screenshot?

**Comment by uchiaTziki** [comment]
attached screenshots... 
1. presenting modal with image picker.
2. picked image
3. present crop controller from that cell
4. after cancelling, the dismissing cause the image picker to vanish (and you're seeing the initial presenting controller - I coloured the controller in white)

Hope that makes things cleared..
![4](https://user-images.githubusercontent.com/20887454/62831971-44309f00-bc30-11e9-8104-6f261a0b0e2d.PNG)
![3](https://user-images.githubusercontent.com/20887454/62831972-44309f00-bc30-11e9-8940-cf865e9031f7.PNG)
![Uploading 2.PNG…]()
![Uploading 1.PNG…]()

**Comment by TimOliver** [comment]
Hmm, so you're presenting it inside another view controller getting presented?

I wonder if this dismiss code is ending up dismissing the wrong view controller then:

https://github.com/TimOliver/TOCropViewController/blob/8770be8407540f5873fe714870771cea8a382ac7/Objective-C/TOCropViewController/TOCropViewController.m#L895

Can you please try changing the code in there and seeing if you can find a combination that fixes it?

Without being able to test it myself, there's nothing I can do from my end. Sorry!

**Comment by uchiaTziki** [comment]
I tried almost anything, but I will try again..
The controller in the back is not being dismissed.. it's view is being removed (which is weird).

**Comment by damozhang** [comment]
Did you try these code before presenting cropViewController?

```
if #available(iOS 13.0, *) {
 cropViewController.modalPresentationStyle = .fullScreen
}
```

**Comment by uchiaTziki** [comment]
Tried it this morning. It does solve the bug but the animation is the default modal animation (from bottom to top). 
The special animation from view/frame doesn't work with this solution

**Comment by LaurentiuUngur** [comment]
@uchiaTziki did you found any fix ?

**Comment by TimOliver** [comment]
Is this reproducible in the sample app at all? If so, I can potentially have a look at it.

The way modal presentations work in iOS 13 changed from previous versions of iOS. But I haven't had a good chance to work out exactly what we need to do to update the code for that yet.

**Comment by LaurentiuUngur** [comment]
As far as I can see, it's not working properly when you have the following:

- Navigation Controller **(A)** **pushes** a View Controller **(B)**
-  View Controller **(B)** **presents** View Controller **(C)** as **form sheet**
- View Controller **(C)** **presents** **CropViewController**
- Tap **Cancel** or **Done** on **CropViewController** and you'll can spot the bug.

**Comment by uchiaTziki** [comment]
Sorry I was on vacation.


> Did you try these code before presenting cropViewController?
> 
> ```
> if #available(iOS 13.0, *) {
>  cropViewController.modalPresentationStyle = .fullScreen
> }
> ```

Using that (@TimOliver ) in the latest Xcode beta did the trick. So the transition animation is working for me now.

**Comment by georgealegre** [comment]
This worked for me as well.

**Comment by jangsy7883** [comment]
``` Swift
 func cropViewController(_ cropViewController: CropViewController, didFinishCancelled cancelled: Bool) {
        //This code fails on iOS13.
        //cropViewController.dismiss(animated: true)

        let viewController = cropViewController.children.first!
        viewController.modalTransitionStyle = .coverVertical
        viewController.presentingViewController?.dismiss(animated: true, completion: nil)
    }
```

This method works well on iOS 13.
I don't know why.

I expect you to find a better way.

https://www.dropbox.com/s/wyzc8errinfap1f/TestCrop.zip?dl=0

**Comment by TappediOS** [comment]
Hi, I meet the bug on iOS 13.1

if I use 
`dismiss(animated: true, completion: nil)`
on iOS 13.1,

All ViewController disappear.

<img width="1520" alt="crop" src="https://user-images.githubusercontent.com/42542860/64938262-41ad0f00-d898-11e9-9535-1288601d1675.png">

I don't know why.

**Comment by oblq** [comment]
I just had to set `cropViewController.modalTransitionStyle = .coverVertical` as a workaround.

**Comment by gask** [comment]
Same thing as @oblq here.

**Comment by partriv** [comment]
> Did you try these code before presenting cropViewController?
> 
> ```
> if #available(iOS 13.0, *) {
>  cropViewController.modalPresentationStyle = .fullScreen
> }
> ```

Fixed my issue, thank you

**Comment by malikmani** [comment]
> if #available(iOS 13.0, *) {
>  cropViewController.modalPresentationStyle = .fullScreen
> }

Thanks. This fixed my issue on iOS 13

**Comment by chitranjank** [comment]
after crop image , dismissing the crop controller, at the end of the transition the navigation controller view is being removed and not displayed 

please help me out

**Comment by ibdman1991** [comment]
> after crop image , dismissing the crop controller, at the end of the transition the navigation controller view is being removed and not displayed
> 
> please help me out

Hi! Just set cropViewController.dismiss(animated: false, completion: nil) instead of cropViewController.dismiss(animated: true, completion: nil)

**Comment by GuillermoCiqueHolvi** [comment]
I've been trying to solve this problem too. As @ibdman1991 says, disabling the animation works but it just looks bad. I think removing the `transitioningDelegate` could be a nicer solution, not sure how impactful this would be though.
```swift
    let cropViewController = CropViewController(image: image)
    cropViewController.modalTransitionStyle = .crossDissolve
    cropViewController.transitioningDelegate = nil
```

**Comment by mitalskhandhar** [comment]
if #available(iOS 13.0, *) {
 cropViewController.modalPresentationStyle = .fullScreen
}

This issue is solved in iOS 13 and above Thanks

**Comment by ashwaniInsigma** [comment]
if #available(iOS 13.0, *) {
            cropViewController.modalPresentationStyle = .fullScreen
           // cropViewController.modalTransitionStyle = .crossDissolve
            cropViewController.transitioningDelegate = nil
        }
Working for me

**Comment by TimOliver** [comment]
Uhh. Okay. Since explicitly setting the presentation style to full screen seemed to fix it for the majority of people, I'm closing this off. :)

If you have a specific use-case that requires an explicitly different modal presentation style, please open a new issue and explain very carefully what you're trying to do. Please bear in mind, if it's a bug in iOS 13, there might not be a lot we can do here.

Thanks!
